### ISSUE TITLE
Chrome and Opera fail to load subtitles and captions

### ISSUE THREAD

**Issue by hirvinen** [videojs/video.js#2689]
At least Opera 32.0 and Chrome 45 fail to load subtitles / captions

See example at http://docs.videojs.com/docs/examples/simple-embed/index.html
Open browser console, then click play. Notice that the control bar does not have caption and subtitle controls.
Verify they are accessible by the browser:
```
// hide video.js controls
videojs.getPlayers()['example_video_1'].controlBar.hide();

// show browser controls
document.getElementById('example_video_1_html5_api').controls = true;

// list text tracks seen by video.js
videojs.getPlayers()['example_video_1'].textTracks();
// output on Chrome: TextTrackList {tracks_: Array[0], vdata1444313627889: 15}

// display captions
document.getElementById('example_video_1_html5_api').textTracks[0].mode = 'showing';
// display subtitles
document.getElementById('example_video_1_html5_api').textTracks[1].mode = 'showing';
```
Browser controls can also toggle display of one of these text tracks.

**Comment by nik0s100** [comment]
Same Here too...

**Comment by chemoish** [comment]
Verified that `videojs('example_video_1')` works, but `data-setup={}` does not.

**Comment by mauritznilsson** [comment]
Having the same issue here, the subtitles are shown, but not with the correct styling and no subtitle menu is showing. The problem does not occur with Firefox.

```html
<script>videojs('example_video_1')</script>
```

Has no change what so ever for me.

**Comment by chemoish** [comment]
I believe there is a couple of outstanding issues that doesn't use native captions for FF. So this is probably why it doesn't effect it.

**Comment by gkatsev** [comment]
At first I thought it might be a cross-origin issue but the captions in the demo-page are on the same domain. I'll have to investigate.

**Comment by platbr** [comment]
I have the same problem here with Chrome 46.0.2490.80 (64-bit) on OSX.

**Comment by osanwe** [comment]
When I've changed setup data from
```javascript
var video_setup = { 'controls': true,
                    'preload': 'none',
                    'playbackRates': [0.5, 0.75, 1, 1.25, 1.5, 1.75, 2, 3],
                    'poster': getPosterUrl(jsonObject),
                    'techOrder': [ 'html5', 'flash' ],
                    'tracks': [
                        { src: 'video/0546_prepod_01H.vtt', kind: 'subtitles', srclang: 'ru', label: 'Русский' },
                    ],
                    'plugins': {
                        videoJsResolutionSwitcher: {
                            default: 'low',
                            dynamicLabel: true
                        }
                    }}
```
to
```javascript
var video_setup = { 'controls': true,
                    'preload': 'none',
                    'playbackRates': [0.5, 0.75, 1, 1.25, 1.5, 1.75, 2, 3],
                    'poster': getPosterUrl(jsonObject),
                    'techOrder': [ 'html5', 'flash' ],
                    'plugins': {
                        videoJsResolutionSwitcher: {
                            default: 'low',
                            dynamicLabel: true
                        }
                    }}
```
and added the text track in the initial function
```javascript
player.addRemoteTextTrack({ src: 'video/0546_prepod_01H.vtt', kind: 'subtitles', srclang: 'ru', label: 'Русский' });
```
the subtitles switcher appeared in control bar but I do not see subtitles when the video plays. In FF all works well.

**Comment by izzat4441** [comment]
Video.js with hls and vast preroll doesn't works. Occured cors error. But i configure allow * on wowza media streamer and each web server. Doesn't work. Who can help me? Thanks

**Comment by gkatsev** [comment]
@izzat4441 if you have a particular issue, please open a separate issue about it and not post on a completely unrelated existing issue.

**Comment by izzat4441** [comment]
oh sorry. Thank u

**Comment by osanwe** [comment]
Fix was found. After adding

```
'html5': {
    nativeTextTracks: false
}
```

to video setup subtitles started to work.

With Google Chrome there is another problem. It does not load subtitles from localhost. You can easily use remote subtitles file (i.e. from dropbox) or start Google Chrome with `--disable-web-security` key.

**Comment by STPo** [comment]
Anything new on this issue?
I'm having the same problem with Win7 / Chrome 47.0.2526.106 m and VideoJS 5.4.4.

**Comment by mkody** [comment]
I can confirm that @osanwe's fix works.

**Comment by CoWinkKeyDinkInc** [comment]
I was able to somehow get both the CC and Subtitles button to appear by just fiddling with the HTML, but I can't figure out how I did it again.  The buttons just appeared and I could switch between them.  The subtitles do appear by using the `default` option except now any caption that is supposed to appear less than 20 seconds into the video does not appear.

**Comment by jaythespacehound** [comment]
The following works to get subtitles working on chrome using native captions:
html:
```html
<video id="mov" width="860" height="484" class="video-js vjs-default-skin" 
    controls preload="auto">
</video>
```

js:
```javascript
function createPlayer(src, subs){
      video=document.getElementById("mov");
      var source = document.createElement("source");
      source.setAttribute("src", src);
      source.setAttribute("type","video/mp4");
      var sub = document.createElement("track");
      sub.setAttribute("label", "English");
      sub.setAttribute("kind", "subtitles");
      sub.setAttribute("srclang", "en");
      sub.setAttribute("src", subs);
      video.appendChild(source);
      video.appendChild(sub);
      videojs("mov",{},function(){
      //player is initialised
      });
      }
createPlayer("pathtomediafile","pathtosubfile");
```

Kind of ugly workaround but there you go.
There is a slightly separate issue though in which in firefox, using the above, no track element is created by video.js. Subtitles still magically work but are non-native and video.textTracks is empty (which is bad when you're trying to edit them on the fly)

**Comment by mcbanderson** [comment]
@osanwe's fix worked for me as well.

**Comment by cubeallspark** [comment]
Hello everyone,

I am noob in all of this, can you please post full code example, how you load the player with html and also how you write the script? Thanks
@osanwe

**Comment by mauritznilsson** [comment]
Simply add: `"html5": { nativeTextTracks: false }`
To the option of the VideoJS element.

You can read about adding options to VideoJS here:
[http://docs.videojs.com/docs/guides/options.html](url)

If you for some reason don't figure it out from that, here is the full code that I use:
`var player = videojs("mainVideo", {
  "html5": {
    nativeTextTracks: false
  }
});`

Hope you get it working,
@cubeallspark

**Comment by cubeallspark** [comment]
I got it, thank you very much, I appreciate your help :)


@mauritznilsson @osanwe

**Comment by MCGallaspy** [comment]
Just encountered this same issue on Chrome 48, too. Not sure I understand the underlying cause -- is it lack of vendor support for some html5 feature, or a bug in video.js?

**Comment by alobimday** [comment]
Caption will not work on iOS if nativeTextTracks = false

**Comment by GuilhermeGuiuler** [comment]
Personal someone could help me in chrome not appear to change the subtitles, please help me

**Comment by surikat1978** [comment]
I have the same problem! No captions window in the browsers Google Chrome, Opera, Yandex, Amigo, and Yandex !
Only FireFox running a window , but does not work default captions. Tested on version 5.8.0 ; 5.8.2; 5.8.3 // 
Help advice, what to do?

**Comment by surikat1978** [comment]
Bug will be removed?

**Comment by surikat1978** [comment]
button was hidden.    http://2sub.ru/animation-cartoons/18-bambi-2-bembi-2-2006.html
 Apparently she was hiding a gun when there are no options. And options in chrome physically there. The fox option - in the code page list li, li, li, li. The chromium is only one and it is li captions off. In my browser so.
And I just did a button visible, adding

`.vjs-captions-button {display: block important;!}`

But the options out there still do not physically in the code.

Chrome = = Yandex Opera

no difference, in my code only :

```
<div class="vjs-menu" role="presentation"><ul class="vjs-menu-content" role="menu">
<li class="vjs-menu-item vjs-selected" tabindex="-1" role="menuitemcheckbox" aria-live="polite" aria-checked="true">captions off<span class="vjs-control-text">, selected</span></li>
</ul></div>
```

and all. Here I have copied all the code, you see only one point here.
but the FF, compare

```
<li aria-live="polite" role="menuitem" tabindex="-1" class="vjs-menu-item vjs-texttrack-settings">captions settings<span class="vjs-control-text">, opens captions settings dialog</span></li>
<li aria-checked="true" aria-live="polite" role="menuitemcheckbox" tabindex="-1" class="vjs-menu-item vjs-selected">captions off<span class="vjs-control-text">, selected</span></li>
<li aria-checked="false" aria-live="polite" role="menuitemcheckbox" tabindex="-1" class="vjs-menu-item">Double<span class="vjs-control-text"> </span></li>
<li aria-checked="false" aria-live="polite" role="menuitemcheckbox" tabindex="-1" class="vjs-menu-item">English<span class="vjs-control-text"> </span></li>
<li aria-checked="false" aria-live="polite" role="menuitemcheckbox" tabindex="-1" class="vjs-menu-item">Russian<span class="vjs-control-text"> </span></li>
```

**Comment by gkatsev** [comment]
Hopefully, this is fixed by @mister-ben in #3212. It's released in 5.8.7, please try it out and let us know!

**Comment by digibill** [comment]
I have the same issue on Chromium Version 80.0.3987.100 (Official Build) snap (64-bit), running on Ubuntu 19.10
